### PR TITLE
Document configuration in Console

### PR DESCRIPTION
**Pull request by emiaj**
Add docs for Doc Gen configuration in Logic.

Add remarks for requirements of SharePoint Client Credentials.

### CODE REVIEW

File: docs/Configuration/Configuration.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+### Configuration Settings
+
+![Image of Logic Console Create Document Generation Configuration - Settings](./assets/image004.jpg)
+
+- Configuration name:  
+_(Required)_ The configuration name, any non-empty string is valid.
+
+- ASPOSE license switch:  
+_(Required)_ In order to use `Logic Document Generation` your organization needs to have a valid ASPOSE license. Toggle this switch on to acknowledge that.
```
Comment (base2browns) [inline docs/Configuration/Configuration.md]: probably should say team rather than organisation

File: docs/Configuration/Configuration.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+- Levels:
+_(Optional)_ `Logic Document Generation` allows you to consume templates stored in multiple storages, it uses a hierarchical organization where each _**level**_ down the hierarchy is able to override a template found on its parent _**level**_.  
+By default this screen is pre-populated with two levels (Customer and Department), feel free to modify this to suit your needs.  
+A configuration environment can have zero, one or multiple levels.
```
Comment (base2browns) [inline docs/Configuration/Configuration.md]: I'd probably drop the zero (even though it's true). what's the point without a master level?

File: docs/Configuration/Configuration.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+### Master Storage Settings
+
+- Template provider:  
+_(Required)_ The template storage provider where the master templates are stored.  
+`Logic Document Generation` currently supports two storage providers: `SharePoint 365 Online` and `Azure Blob Storage`.
+
+#### SharePoint settings
+![Image of Logic Console Create Document Generation Configuration - SharePoint Settings](./assets/image005.jpg)
+
+`Logic Document Generation` requires a valid SharePoint Client Credential and a Group name to retrieve templates from it.
```
Comment (base2browns) [inline docs/Configuration/Configuration.md:50]: How does someone create these?

File: docs/Configuration/Configuration.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+- Group Name:  
+_(Required)_ The SharePoint group name where the templates are stored.
+
+- Client Secret:  
+_(Required)_ The Client Secret from your SharePoint Client Credentials.
+
+#### Azure Blob Storage settings
+![Image of Logic Console Create Document Generation Configuration - Azure Blob Storage Settings](./assets/image006.jpg)
+
```
Comment (base2browns) [inline docs/Configuration/Configuration.md:67]: How does someone create these? I assume create a blob storage, add a service principal, copy these details from here to here ...

**Review by base2browns — CHANGES_REQUESTED** [review]

File: docs/Configuration/Configuration.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
+﻿# KMD Logic Document Generation Configuration
+
+To configure a Logic Document Generation environment, you need to access [Logic Console](https://console.kmdlogic.io/) to do so.
```
Comment (danielhazel) [inline docs/Configuration/Configuration.md]: "to do so" can be removed.

File: docs/Configuration/Configuration.md (hunk 2 above)
Comment (danielhazel) [reply docs/Configuration/Configuration.md]: If its zero, one or multiple levels, we're really only ruling out negatives :)
So we could maybe even lose the whole line.

File: docs/Configuration/Configuration.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+  - Edit entry (pencil icon):  
+  Allows you to edit the configuration entry.  
+  It will open a pop-up dialog like so:  
+![Image of Logic Console Create Document Generation Configuration - Edit pop up](./assets/image009.jpg)
+
+  - Add child entry (plus icon):  
+  Adds a child entry to the configuration entry.  
+  It will open a pop-up dialog like so:  
+  ![Image of Logic Console Create Document Generation Configuration - Add pop up](./assets/image009.jpg)
```
Comment (danielhazel) [inline docs/Configuration/Configuration.md]: This is the same image as the Edit pop up.

File: docs/Configuration/Configuration.md (hunk 2 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md]: Not having levels means you can only have a Master entry and Master cannot have children entries.
Say you have only 1 level, that means Master can have children entries.
So zero is still a valid number of child levels.

File: docs/Configuration/Configuration.md (hunk 3 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md:50]: I thought as a general guideline we wouldn't explain external services.

Would putting a link to the official documentation be good enough?

File: docs/Configuration/Configuration.md (hunk 2 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md]: @danielhazel I agree, removing the paragraph may be better.

File: docs/Configuration/Configuration.md (hunk 3 above)
Comment (danielhazel) [reply docs/Configuration/Configuration.md:50]: Hehe.
I think my most persuasive argument for saying "not me" about this doco was that I was the only one that didn't have sharepoint access so I was the only one that couldn't build the sharepoint walkthru.
:)

File: docs/Configuration/Configuration.md (hunk 1 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md]: Fixed in 3e485059348ca4959c60435516832dab8ec7c166

File: docs/Configuration/Configuration.md (hunk 5 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md]: Fixed in deeb2f7950fd5fcc6858b8763b0c16761a2c9d91

File: docs/Configuration/Configuration.md (hunk 6 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md]: Fixed in 2311c5eccc154b7b0a8a079e6daafee382936454

File: docs/Configuration/Configuration.md (hunk 2 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md]: Fixed in 100845b4e1e48294a117fb0efdc084d7ba920895

File: docs/Configuration/Configuration.md (hunk 3 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md:50]: I will address this in a separate PR.

File: docs/Configuration/Configuration.md (hunk 4 above)
Comment (emiaj) [reply docs/Configuration/Configuration.md:67]: I will address this in a separate PR.

**Pull request merged**